### ISSUE TITLE
False positive (warning: first argument to metaclass `__new__` should be `mcs`, but `cls` is suggested.

### ISSUE THREAD

**Issue by patrick-kidger** [microsoft/pyright#4736]
```python
class Meta(type):
    def __new__(mcs, name, bases, dct):
        ...
```
produces a `warning: __new__ override should take a "cls" parameter (reportSelfClsParameterName)`.

However in a metaclass then usually the first parameter would be called something like `mcs`.

In particular `cls` is a confusing recommendation, given that in the body we'd usually then write something like `cls = super().__new__(mcs, name, bases, dct)`.

Python 3.8.0.
Pyright 1.1.296 run from the command line.

**Comment by erictraut** [comment]
If you want to use different parameter naming conventions, you can disable the `reportSelfClsParameterName` check in pyright. This check is probably better handled by a linter than a type checker anyway.

**Comment by erictraut** [comment]
On second thought, I've seen both `mcs` and `mcls` used in metaclasses. I've updated the check to allow these. If you want to enforce more specific naming conventions for parameters, I recommend using a linter.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.297, which I just published. It will also be included in a future release of pylance.
